### PR TITLE
swarm: propagate ctx to internal apis

### PR DESCRIPTION
**Pull request by nonsense**
I think this will reduce conflicts between the chunker refactor and the introduction of tracing information within the context.

### CODE REVIEW

File: swarm/api/http/server.go [hunk 1]
```diff
@@ -1011,6 +1012,8 @@ func (b bufferedReadSeeker) Seek(offset int64, whence int) (int64, error) {
 }
 
 func (s *Server) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	ctx := context.TODO()
+
 	defer metrics.GetOrRegisterResettingTimer(fmt.Sprintf("http.request.%s.time", r.Method), nil).UpdateSince(time.Now())
 	req := &Request{Request: *r, ruid: uuid.New()[:8]}
```
Comment (nonsense) [inline swarm/api/http/server.go:1018]: Note that even if we don't go through with Opentracing, at least `ruid` belongs to the context and not to the `Request` obj. We would be able to propagate it much further with context and get more meaningful traces through this correlation id.

File: swarm/api/http/server.go [hunk 2]
```diff
@@ -120,7 +121,7 @@ type Request struct {
 
 // HandlePostRaw handles a POST request to a raw bzz-raw:/ URI, stores the request
 // body in swarm and returns the resulting storage address as a text/plain response
-func (s *Server) HandlePostRaw(w http.ResponseWriter, r *Request) {
+func (s *Server) HandlePostRaw(ctx context.Context, w http.ResponseWriter, r *Request) {
```
Comment (janos) [inline swarm/api/http/server.go:124]: Maybe to use Context from *Request, instead to pass another one?

File: swarm/api/http/server.go [hunk 3]
```diff
@@ -1011,6 +1012,8 @@ func (b bufferedReadSeeker) Seek(offset int64, whence int) (int64, error) {
 }
 
 func (s *Server) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	ctx := context.TODO()
```
Comment (janos) [inline swarm/api/http/server.go:1015]: I think that this ctx should not be created, as the context propagation for http is done with Request.

**Review by janos — COMMENTED** [review]
My comments are related to context that already is in Request object and that the new context instance should not be passed to http Handlers. Otherwise, LGTM.

**Review by gbalint — APPROVED** [review]
Besides @janos 's comment on the context of the Request, LGTM

**Pull request merged**